gapis/api/controlFlowGenerator/linear_control_flow_generator.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controlFlowGenerator
+
+import (
+	"context"
+
+	"github.com/google/gapid/gapis/api/commandGenerator"
+	"github.com/google/gapid/gapis/api/transform2"
+)
+
+type linearControlFlowGenerator struct {
+	transforms   []transform2.Transform
+	cmdGenerator commandGenerator.CommandGenerator
```

Comment:
The TransformChain should be constructed with the slice of transforms and the command generator. The transform chain should then be passed into the control flow generator's constructor.